File: Uni/sb223ce_assign2/ReverseOrder.java
package sb223ce_assign2;

import java.util.Scanner;
import java.util.ArrayList;

public class ReverseOrder {

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		System.out.println("Enter a positive integer. End by giving negative integer.");
		ArrayList<Integer> number = new ArrayList<Integer>();

		/*As we do not know how many integers user will input, so we use 'do'
		 * statement and save the integers in array list until negative integer
		 * has been provide.*/

		int input;
		do {

			input = scan.nextInt();

			if (input >= 0) {
				number.add(input);
			}
		} while (input > -1);

		System.out.println("PostiveInteger: " + (number.size()));

		System.out.print("In reverese order: ");

		/*For reversing we run loop from last index.*/

		for (int i = number.size() - 1; i >= 0; i--) {

			System.out.print((number.get(i)) + " ");
			/* used space to differentiate the integers */

		}

		scan.close();
	}

}

File: Assignment 2/C# version/ReverseOrder.cs
using System;
using System.Collections;
using System.Runtime.Remoting.Metadata.W3cXsd2001;
using System.Collections.Generic;

class ReverseOrder
{
	public static void Main(string[] args)
	{
		List<int> list = new List<int>();
		Console.WriteLine("Enter positive integers. End by giving a negative integer.");

		int counter = 0;
		int input = 0;

		while (input >= 0)
		{
			Console.Write("Integer " + (++counter) + ": ");
			input = Convert.ToInt32(Console.ReadLine());
			if (input >= 0)
			{
				list.Add(input);
			}
		}

		Console.WriteLine("Number of positive integers: " + list.Count);
		Console.Write("In reverse order: ");

		for (int i = list.Count - 1; i >= 0; i--)
		{
			Console.Write(list[i] + "  ");
		}
	}
}